src/main/java/org/jenkinsci/main/modules/instance_identity/InstanceIdentity.java

```diff
@@ -53,19 +54,32 @@ public InstanceIdentity(File keyFile) throws IOException {
             keys = read(new FileReader(oldKeyFile), gen);
             write(keys, keyFile);
             Util.deleteFile(oldKeyFile);
-        } else if (keyFile.exists()) {
-            byte[] enc = FileUtils.readFileToByteArray(keyFile);
+        } else {
+            Reader reader = getFileReader(keyFile);
+            if (reader!=null) {
+                keys = read(reader, gen);
+            } else {
+                gen.initialize(2048, new SecureRandom()); // going beyond 2048 requires crypto extension
+                keys = gen.generateKeyPair();
+                write(keys, keyFile);
+            }
+        }
+    }
```

Comment:
In fact, is the order correct here i.e. it tries reading from the `oldKeyFile` first but maybe it should start with `keyFile` and fall back to `oldKeyFile`?